app/models/user.rb

```diff
@@ -14,10 +14,11 @@ class User < ApplicationRecord
   accepts_nested_attributes_for :permissions, allow_destroy: true
 
   before_validation :set_uid, on: :create
+  before_save :set_sort_name
```

Comment:
I like the idea, especially since I can't think of any cases where we only show a partial name, but that would mean that we need to enter a sort name every time we set or update the full_name, which I don't love.

Reply:
Why couldn't we just use the full_name to sort?

Reply:
You mean sort lists of users by their first name?  Based on existing code, I thought you wanted to sort by last name.

Reply:
I think I just choose last name randomly or because it's what was done in Hyacinth 2. We could ask HUG, but I am not sure they would have a strong opinion. When I have looked for a name, I usually go by first name (sometimes i don't know someone's last name lol). If we just have a full name field, we can avoid having sort_name, first_name, last_name and a middle_name field.

Reply:
Haha, I sometimes do the first name lookup for the same reason.  I'm good with going with full name for now then, and sorting by whatever the full name starts with.  We can always re-add the other fields later if needed, or maybe better yet make sure that our UIs allow typeahead name search.  Do you mind if we do that in a PR that immediately follows this one?  This one's getting pretty big and the name stuff touches several parts of the app.

Reply:
A separate PR would be great!